api/airports/airports.controller.js

```diff
@@ -2,26 +2,48 @@ const mongoose = require('mongoose');
 const Airports = mongoose.model('airports');
 const boom = require('boom');
 
-exports.search =(req, res, next) => {
+exports.search = (req, res, next) => {
 
+    let search = req.query.data;
+    let search2 = req.query.data2;
 
-}
+    if (search.length <= 2 || search2.length <= 2) {
+       return next(boom.badImplementation("Please provide atleast 3 letters to search"));
+    } 
 
+        Airports.find({name: {$regex: search,$options: 'i'}, iata:{$regex: search2,$options: 'i'} })
 
```

Comment:
I have fixed the code you can review it..
https://github.com/khurrumqureshi/test-portal/tree/TP-2